### PR TITLE
change the `ResourceGroupID` to optional on alibabacloud provider

### PR DESCRIPTION
**Pull request by BSWANG**
The `ResourceGroupID` is not required, so change the `ResourceGroupID` to optional.

### CODE REVIEW

File: config/v1/types_infrastructure.go [hunk 1]
```diff
@@ -665,10 +665,8 @@ type AlibabaCloudPlatformStatus struct {
 	// +required
 	Region string `json:"region"`
 	// resourceGroupID is the ID of the resource group for the cluster.
-	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:Pattern=`^rg-[0-9A-Za-z]+$`
-	// +required
-	ResourceGroupID string `json:"resourceGroupID"`
+	// +optional
+	ResourceGroupID string `json:"resourceGroupID,omitempty"`
```
Comment (sttts) [inline config/v1/types_infrastructure.go:669]: omitempty and required doesn't make sense. Why do you remove the kubebuilder pattern tag? It must stay.

Comment (sttts) [reply config/v1/types_infrastructure.go:669]: Moving from required to optional break downgrades. Is that considered?

Comment (bparees) [reply config/v1/types_infrastructure.go:669]: > Moving from required to optional break downgrades. Is that considered?

this is the first release of the api

**Pull request merged**